test/cace/test_amm_typing.c

```diff
@@ -288,3 +288,33 @@ void test_amm_type_convert_ident(const char *inhex, const char *expecthex)
     const amm_type_t *type = amm_type_get_builtin(ARI_TYPE_IDENT);
     check_convert(type, inhex, expecthex);
 }
+
+TEST_CASE("820C1A2B450625", "820C1A2B450625") // ari:/TP/20230102T030405Z
+TEST_CASE("82041864", "820C1864") // ari://INT/100 -> ari://TP/100
+TEST_CASE("8208FB40593F34D6A161E5", "820C82231A000F68D4") // ari://REAL64/100.9876
+TEST_CASE("8209FA5F000000", NULL) // ari://REAL64/0x7FFFFFFFFFFFFFFFF --> exceeds INT64 MAX
+TEST_CASE("820D1864", "820C1864") // ari://TD/100 -> ari://TP/100
```

Comment:
This one should actually fail. The same with the other direction to TD. Both can convert from primitive, int, float values but not between each other. I think the negative int case below should succeed.

Reply:
Thanks for clarifying, wasn't sure if we wanted to allow TD/TP conversions. Updated accordingly and also fixed the negative int case as well.